hasher-matcher-actioner/hmalib/common/timebucketizer.py

```diff
@@ -174,8 +176,61 @@ def get_records(
 
         for file in file_list:
             with open(file, "r") as my_file:
-                content_list.extend(
-                    list(map(type_class().from_csv, csv.reader(my_file)))
-                )
+                content_list.extend(list(map(type_class.from_csv, csv.reader(my_file))))
 
         return content_list
+
+    @staticmethod
+    def squash_directory(directory_path):
+        file_list = []
+        for file in os.listdir(directory_path):
+            file_path = os.path.join(directory_path, file)
+            if file.startswith("squash"):
+                logger.info("Directory (%s) already squashed.", directory_path)
+                return
+            elif os.path.isfile(file_path):
+                file_list.append(file_path)
+
+        with open(
+            os.path.join(directory_path, "squash_" + str(uuid.uuid1())) + ".csv",
+            "w",
+        ) as new_file:
+            for file in file_list:
+                logger.info("getting content of: %s", file)
+                with open(file, "r") as reader:
+                    new_file.write(reader.read())
+                logger.info("removing file: %s", file)
+                os.remove(file)
+        logger.info("successfully squashed: %s", directory_path)
+
+    @staticmethod
+    def squash_content(
+        type: str,
+        storage_path: str,
+        bucket_width: datetime.timedelta,
+        max_bucket_start_time: datetime.datetime,
+        min_bucket_start_time: datetime.datetime,
+    ):
+        until_nearest = TimeBucketizer._calculate_bucket_endpoints(
+            max_bucket_start_time, bucket_width
+        )[1]
+        since_nearest = TimeBucketizer._calculate_bucket_endpoints(
+            min_bucket_start_time, bucket_width
+        )[0]
+
+        if until_nearest > datetime.datetime.now() - bucket_width:
+            logger.info(
+                "Start time indicates nearest bucket is not ready for squashing"
+            )
+            return
+
+        while until_nearest <= since_nearest:
```

Comment:
This is weird. Why is until < since at all? 

since is always < until, right?

Reply:
I'm worried we are not even entering this loop because until > since. 

This means that your test (which is passing) is not capturing all the behavior. The test is counting and comparing records, but that is not enough.

Internally, whether the number of files reduced at all is not being checked for. Your unit test should do that.